internal/servegit/serve.go

```diff
@@ -130,6 +130,49 @@ func (s *Serve) handler() http.Handler {
 	})
 }
 
+func isDir(path string) bool {
+	if fi, err := os.Stat(path); err != nil || !fi.IsDir() {
+		return false
+	}
+
+	return true
+}
+
+// Checks if git thinks the given path is a valid .git folder for a repository
+func isBareRepo(path string) bool {
+	if !isDir(path) {
+		return false
+	}
+
+	c := exec.Command("git", "--git-dir", path, "rev-parse", "--is-bare-repository")
+	c.Dir = path
+	out, err := c.CombinedOutput()
+
+	if err != nil {
+		return false
+	}
+
+	return string(out) != "false\n"
+}
+
+// Check if git thinks the given path is a proper git checkout
+func isGitRepo(path string) bool {
+	if !isDir(path) {
+		return false
+	}
```

Comment:
I wanted to suggest to pull `isDir(path)` out of these two functions, but then I saw these 3 lines:

https://github.com/sourcegraph/src-cli/pull/639/files#diff-2d1e7e4eecf58c3cd47322de771f3e05f7eba484526fe2e85d35289c2cb8fcbfR192-R194

So, I don't think the `isDir` check (here and in the other function) is even necessary, right?

Reply:
it is a quick, cheaper way to abort early. I could imagine that with lots of repositories (or folders) this could consume some time and this would at least ensure that we aren't forking/exec'ing very often.

Reply:
Yes, but:

1. We do `if !isDir(path) { return false }` _twice_ in both of these functions. We can pull it out and do it once.
2. We _already_ check whether the path is a directory by checking `fi.IsDir` in the lines above.

Reply:
I removed the duplicate checks now.